src/data/nat/pairing.lean

```diff
@@ -129,6 +129,32 @@ begin
     exact le_trans h₁ (nat.le_add_left _ _) }
 end
 
+theorem mkpair_lt_max_succ_sq (m n : ℕ) : mkpair m n < (max m n + 1) ^ 2 :=
+begin
+  rw mkpair,
+  split_ifs,
+  { rw [max_eq_right h.le, add_sq, add_assoc, sq, add_lt_add_iff_left, mul_one,
+      two_mul, add_assoc],
+    exact h.trans_le (self_le_add_right n _) },
+  { rw not_lt at h,
+    rw [max_eq_left h, add_sq, add_assoc, add_assoc, sq, add_lt_add_iff_left, mul_one,
+      two_mul, add_assoc, add_lt_add_iff_left],
+    exact lt_succ_of_le h }
+end
+
+theorem add_le_mkpair (m n : ℕ) : m + n ≤ mkpair m n :=
```

Comment:
You can also have `max m n ^ 2 + min m n ≤ mkpair m n` if you want.

Reply:
This won't help with my specific application, though. The bounds are pretty fine tuned and hard to change.

Reply:
But can't hurt to have. Let me add it to the PR.